emission/analysis/modelling/tour_model/similarity.py

```diff
@@ -35,11 +35,20 @@ class similarity(object):
     
     def __init__(self, data, radius):
         self.data = data
-        if not data:
-            self.data = []
```

Comment:
So the constructor used to filter the data, but it doesn't any more. So you are changing the functionality of the constructor as part of a refactoring, which is generally a bad idea. A refactoring should change the structure, *not* the functionality.